pkg/hostagent/hostagent.go

```diff
@@ -143,6 +143,12 @@ func (a *HostAgent) Run(ctx context.Context) error {
 		a.emitEvent(ctx, exitingEv)
 	}()
 
+	dnsServer, err := a.StartDNS()
```

Comment:
Could you add the design document to `internal.md`?

Reply:
And IIUC, we are going to let guestagent to listen on 127.X.X.X:53 on the guest and forward it to 192.168.5.2:XXXXX (which is forwarded to 127.0.0.0.1:XXXXX of the host), right?

Reply:
And I guess it is not necessarily to listen on 127.0.0.1:XXXXX of the host.
We can just listen it on an UNIX socket on the host and use QEMU guestfwd. (`-nic user,guestfwd=tcp:192.168.5.2:XXXXX-unix:dns.sock)

Reply:
> And IIUC, we are going to let guestagent to listen on 127.X.X.X:53 on the guest

We will, unless this can be done automatically via `iptables`.

> 192.168.5.2:XXXXX (which is forwarded to 127.0.0.0.1:XXXXX of the host), right?

`192.168.5.2` **is** the gateway/host, so all ports from the host are automatically routable through it without any forwarding.

Reply:
> And I guess it is not necessarily to listen on 127.0.0.1:XXXXX of the host.
> We can just listen it on an UNIX socket on the host and use QEMU guestfwd. (`-nic user,guestfwd=tcp:192.168.5.2:XXXXX-unix:dns.sock)

This is not necessary; it already works without any further configuration.

Reply:
Also this is UDP; and we can't transport UDP over a domain socket (at least I don't think you can; I may be wrong).

Reply:
The guestagent should listen on 53/udp, but we can use arbitrary protocol for the communication between the guestagent and the hostagent.